src/rgw/rgw_redis_driver.h

```diff
@@ -0,0 +1,111 @@
+    virtual std::vector<Entry> list_entries(const DoutPrefixProvider* dpp) override;
+    virtual size_t get_num_entries(const DoutPrefixProvider* dpp) override;
+    //int update_local_weight(const DoutPrefixProvider* dpp, std::string key, int localWeight); // may need to exist for base class -Sam
+
+    /* Partition */
+    virtual Partition get_current_partition_info(const DoutPrefixProvider* dpp) override { return partition_info; }
+    virtual uint64_t get_free_space(const DoutPrefixProvider* dpp) override { return free_space; } // how to get this from redis server? -Sam
+    static std::optional<Partition> get_partition_info(const DoutPrefixProvider* dpp, const std::string& name, const std::string& type);
+    static std::vector<Partition> list_partitions(const DoutPrefixProvider* dpp);
+
+    virtual int initialize(CephContext* cct, const DoutPrefixProvider* dpp) override;
+    virtual int put(const DoutPrefixProvider* dpp, const std::string& key, bufferlist& bl, uint64_t len, rgw::sal::Attrs& attrs) override;
+    virtual int get(const DoutPrefixProvider* dpp, const std::string& key, off_t offset, uint64_t len, bufferlist& bl, rgw::sal::Attrs& attrs) override;
+    virtual int append_data(const DoutPrefixProvider* dpp, const::std::string& key, bufferlist& bl_data) override;
+    virtual int delete_data(const DoutPrefixProvider* dpp, const::std::string& key) override;
+    virtual int get_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& attrs) override;
+    virtual int set_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& attrs) override;
+    virtual int update_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& attrs) override;
+    virtual int delete_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& del_attrs) override;
+    virtual std::string get_attr(const DoutPrefixProvider* dpp, const std::string& key, const std::string& attr_name) override;
+    virtual int set_attr(const DoutPrefixProvider* dpp, const std::string& key, const std::string& attr_name, const std::string& attr_val) override;
+
+    virtual std::unique_ptr<CacheAioRequest> get_cache_aio_request_ptr(const DoutPrefixProvider* dpp) override { return nullptr; }
+    virtual rgw::AioResultList get_async(const DoutPrefixProvider* dpp, optional_yield y, rgw::Aio* aio, const std::string& key, off_t ofs, uint64_t len, uint64_t cost, uint64_t id) override;
+
+    struct redis_aio_handler { 
+      rgw::Aio* throttle = nullptr;
+      rgw::AioResult& r;
+      boost::redis::response<std::string>* resp;
+      bufferlist* bl;
+
+      /* Read Callback */
+      void operator()(boost::system::error_code ec, auto) const {
+	r.result = -ec.value();
+
+	const std::string s = std::get<0>(*resp).value();
+	bl->append(s.c_str(), s.size()); 
+
+	r.data = *bl;
+	throttle->put(r);
+
+	delete resp;
```

Comment:
in general, you should always try to use `std::unique_ptr` and `std::make_unique()` instead of raw `new` and `delete`. memory management is a very common source of bugs, and smart pointers make it much harder to get wrong

in this case, we're not guaranteed that `operator()` will get called for every `redis_aio_handler`. if the io_context stops before it completes, the handler will just destruct. if the `resp` member variable is stored in a unique_ptr, it'll free the memory automatically on destruction